### ISSUE TITLE
Symposium and HTTPS issue

### ISSUE THREAD

**Issue by rdohms** [rdohms/pronto#40]
Symposium has enabled HTTPS.
Pronto v0.4.7 has switched to HTTP url, and that has been added to manifest, but we are still getting CORS errors that need to be looked into.

`XMLHttpRequest cannot load https://symposiumapp.com/oauth/access-token. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'chrome-extension://ceppkbmiglldpkelkicajinlfdkbpihh' is therefore not allowed access.`

**Comment by mattstauffer** [comment]
Every time you try to load, the server is throwing this error:

**The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed. Check the "access token" parameter.**

Important points from the stack trace:

```
Stack trace:
#0 /home/forge/symposiumapp.com/vendor/league/oauth2-server/src/ResourceServer.php(111): League\OAuth2\Server\ResourceServer->determineAccessToken(false)
#1 /home/forge/symposiumapp.com/vendor/lucadegasperi/oauth2-server-laravel/src/Authorizer.php(223): League\OAuth2\Server\ResourceServer->isValidRequest(false, NULL)
#2 /home/forge/symposiumapp.com/vendor/lucadegasperi/oauth2-server-laravel/src/Middleware/OAuthMiddleware.php(72): LucaDegasperi\OAuth2Server\Authorizer->validateAccessToken(false)
#3 [internal function]: LucaDegasperi\OAuth2Server\Middleware\OAuthMiddleware->handle(Object(Illuminate\Http\Request), Object(Closure))
```

Full stack trace:

```
[2016-06-10 13:21:03] production.ERROR: League\OAuth2\Server\Exception\InvalidRequestException: The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed. Check the "access token" parameter. in /home/forge/symposiumapp.com/vendor/league/oauth2-server/src/ResourceServer.php:150
Stack trace:
#0 /home/forge/symposiumapp.com/vendor/league/oauth2-server/src/ResourceServer.php(111): League\OAuth2\Server\ResourceServer->determineAccessToken(false)
#1 /home/forge/symposiumapp.com/vendor/lucadegasperi/oauth2-server-laravel/src/Authorizer.php(223): League\OAuth2\Server\ResourceServer->isValidRequest(false, NULL)
#2 /home/forge/symposiumapp.com/vendor/lucadegasperi/oauth2-server-laravel/src/Middleware/OAuthMiddleware.php(72): LucaDegasperi\OAuth2Server\Authorizer->validateAccessToken(false)
#3 [internal function]: LucaDegasperi\OAuth2Server\Middleware\OAuthMiddleware->handle(Object(Illuminate\Http\Request), Object(Closure))
#4 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(9913): call_user_func_array(Array, Array)
#5 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#6 /home/forge/symposiumapp.com/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#7 /home/forge/symposiumapp.com/vendor/lucadegasperi/oauth2-server-laravel/src/Middleware/OAuthExceptionHandlerMiddleware.php(36): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#8 [internal function]: LucaDegasperi\OAuth2Server\Middleware\OAuthExceptionHandlerMiddleware->handle(Object(Illuminate\Http\Request), Object(Closure))
#9 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(9913): call_user_func_array(Array, Array)
#10 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#11 /home/forge/symposiumapp.com/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#12 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(13362): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#13 [internal function]: Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#14 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(9913): call_user_func_array(Array, Array)
#15 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#16 /home/forge/symposiumapp.com/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#17 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(11908): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#18 [internal function]: Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#19 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(9913): call_user_func_array(Array, Array)
#20 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#21 /home/forge/symposiumapp.com/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#22 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(13101): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#23 [internal function]: Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
#24 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(9913): call_user_func_array(Array, Array)
#25 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#26 /home/forge/symposiumapp.com/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#27 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(13038): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#28 [internal function]: Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
#29 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(9913): call_user_func_array(Array, Array)
#30 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#31 /home/forge/symposiumapp.com/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#32 [internal function]: Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#33 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(9898): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#34 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(8179): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#35 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(8170): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#36 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(8160): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#37 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(2416): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#38 [internal function]: Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#39 /home/forge/symposiumapp.com/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(52): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#40 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(3271): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#41 [internal function]: Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure))
#42 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(9913): call_user_func_array(Array, Array)
#43 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#44 /home/forge/symposiumapp.com/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#45 [internal function]: Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#46 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(9898): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#47 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(2363): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#48 /home/forge/symposiumapp.com/bootstrap/cache/compiled.php(2347): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#49 /home/forge/symposiumapp.com/public/index.php(52): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#50 {main}
```

**Comment by mattstauffer** [comment]
I'm not sure where Pronto is expecting to set the header, but I sniffed it and it's not in the headers:

```
array (
  'accept-language' =>
  array (
    0 => 'en-US,en;q=0.8',
  ),
  'accept-encoding' =>
  array (
    0 => 'gzip, deflate, sdch, br',
  ),
  'dnt' =>
  array (
    0 => '1',
  ),
  'user-agent' =>
  array (
    0 => 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.102 Safari/537.36',
  ),
  'origin' =>
  array (
    0 => 'chrome-extension://ceppkbmiglldpkelkicajinlfdkbpihh',
  ),
  'accept' =>
  array (
    0 => '*/*',
  ),
  'connection' =>
  array (
    0 => 'keep-alive',
  ),
  'host' =>
  array (
    0 => 'symposiumapp.com',
  ),
  'content-length' =>
  array (
    0 => '',
  ),
  'content-type' =>
  array (
    0 => '',
  ),
)
```

and it's not in the URL:

```
https://symposiumapp.com/api/me
```

**Comment by rdohms** [comment]
I'm wondering if we lost something in the switch, cause i'm pretty sure we solved this once before.

**Comment by rdohms** [comment]
Looking here at the : "login" process which is the one failing with that message and the request is proper:

```
POST /oauth/access-token HTTP/1.1
Host: symposiumapp.com
Connection: keep-alive
Content-Length: 255
Accept: */*
Origin: chrome-extension://ceppkbmiglldpkelkicajinlfdkbpihh
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2751.0 Safari/537.36
Content-Type: application/x-www-form-urlencoded; charset=UTF-8
Accept-Encoding: gzip, deflate, br
Accept-Language: en,en-US;q=0.8,pt-BR;q=0.6,pt;q=0.4
```

and the content

```
code: [redacted]
grant_type:authorization_code
redirect_uri:https://ceppkbmiglldpkelkicajinlfdkbpihh.chromiumapp.org/provider_cb
client_id: [redacted]
client_secret: [redacted]
```

Shows as 200 OK without any content and gives out the alert above.

**Comment by mattstauffer** [comment]
OK, I was testing the request-once-logged-in. When I get back in later today I'll take a look at the login request and see if I can debug that.

**Comment by mattstauffer** [comment]
OK, so I just tried it from my machine. I logged out from Pronto using the extension options page.

I hit the login button there and got a popup for authentication. It took me to `/login` and then once I filled that out it took me to `/oauth/authorize`. I approved the OAuth prompt and then the screen disappeared. 

It then returned me to the options panel with nothing changed, no indication of success:

![image](https://cloud.githubusercontent.com/assets/151829/15992970/8a356854-30a7-11e6-87ce-7b3fee865c27.png)

I checked the server logs, and saw that `/oauth/access-token` had been accessed, and a JSON response was returned:

``` json
{"access_token":"my token here","token_type":"Bearer","expires_in":31536000}
```

I don't know how to sniff all of this from the Chrome side, but from the server side I could see that URL getting hit and I can see the response it returned. I didn't do anything special with CORS or anyhting on my machine. 

Chrome Version 51.0.2704.84 (64-bit)
Pronto version 0.4.7
